### PR TITLE
Move width and height to the base renderer class definition.

### PR DESCRIPTION
**Pull request by manthey**
This reduces code duplication.  It has been listed as a TODO in the code for a long time.

### CODE REVIEW

File: src/d3/d3Renderer.js [hunk 1]
```diff
@@ -209,11 +207,10 @@ var d3Renderer = function (arg) {
         width = map.size().width,
         height = map.size().height;
 
-    m_width = width;
-    m_height = height;
-    if (!m_width || !m_height) {
+    if (!width || !height) {
```
Comment (aashish24) [inline src/d3/d3Renderer.js]: in vglRenderer we use m_this.width() to get width and height, why not we use the same here? instead of getting it from map?

Comment (manthey) [reply src/d3/d3Renderer.js]: A few lines later is where the layer's width and height is set *from* these values (`m_this._setWidthHeight(width, height);`), so `width` and `height` are the values that will be set.

Comment (aashish24) [reply src/d3/d3Renderer.js]: I see, I was expecting the flow between vgl and d3 renderers would be same and may be they are similar. so sounds like we are getting the width and height from the map and set it to the renderer in both cases?

Comment (manthey) [reply src/d3/d3Renderer.js]: Yes.

**Pull request merged**